checks/apps/pytorch/pytorch_megatronlm_amd.py

```diff
@@ -0,0 +1,433 @@
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+import os
+import pathlib
+import sys
+
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+sys.path.append(str(pathlib.Path(__file__).parent.parent.parent / 'mixins'))
+
+from container_engine import ContainerEngineMixin  # noqa: E402
+
+
+class PyTorchMegatronLM_AMD(rfm.RunOnlyRegressionTest):
+    num_tasks_per_node = 1
+    default_num_nodes = variable(int, type(None), value=None)
+    time_limit = '30m'
+    megatron_repo = variable(
+        str, value='https://github.com/ROCm/Megatron-LM'
+    )
+    megatron_release = variable(str, value='38fc830')
+    model = parameter(['llama2-7b'])
+    exit_interval = variable(int, value=10)
+    checkpoint_dir = variable(str, type(None), value=None)
+    dataset_cache_dir = variable(str, type(None), value=None)
+    enable_fp8 = variable(bool, value=False)
+    sequence_parallel = variable(bool, value=True)
+    gemm_tuning = variable(bool, value=True)
+    mcore = variable(bool, value=True)
+    conti_params = variable(bool, value=False)
+    fsdp = variable(bool, value=False)
+    use_flash_attn = variable(bool, value=True)
+    rope_fusion = variable(bool, value=True)
+    eval_interval = variable(int, value=5000)
+    save_interval = variable(int, value=5000)
+    eval_iters = variable(int, value=1)
+    recompute = variable(bool, value=False)
+    nccl_debug = variable(bool, value=False)
+    batch_size_per_node = variable(int, value=256)
+    checkpoint_steps = variable(int, value=10)
+    hf_home = variable(
+        str, value=str(pathlib.Path.home() / '.cache' / 'huggingface')
+    )
+    training_steps = variable(int, value=10)
+    wandb_logging = variable(bool, value=False)
+    datasets = variable(str, type(None), value=None)
+
+    configurations = {
+        'llama2-7b': {
+            'num_nodes': 16,
+            'micro_batch_size': 8,
+            'num_layers': 32,
+            'hidden_size': 4096,
+            'ffn_hidden_size': 11008,
+            'num_attention_heads': 32,
+            'num_kv_heads': 32,
+            'sequence_length': 4096,
+            'tensor_model_parallel_size': 4,
+            'pipeline_model_parallel_size': 1,
+            'context_parallel_size': 1,
+            'max_position_embeddings': 32000,
+            'activation': 'swiglu',
+            'optimizer': 'adam',
+            'regularization_args': [
+                '--attention-dropout 0.0',
+                '--hidden-dropout 0.0',
+                '--weight-decay 0.1',
+                '--clip-grad 1.0',
+                '--adam-beta1 0.9',
+                '--adam-beta2 0.95',
+            ],
+            'learning_rate_args': [
+                f'--lr 1.e-4',
+                f'--min-lr 1.e-5',
+                f'--lr-decay-style cosine',
+                f'--lr-warmup-iters 1',
+            ]
+        }
+    }
+
+    sourcesdir = None
+    executable = 'bash'
+
+    tags = {'maintenance', 'production', 'ml'}
```

Comment:
This test takes ~25 min on 16 nodes to run 10 iterations:
```
grep 'throughput per GPU' rfm_job.out |awk '{print $4,$26,$25}'
1/ 25.5 (TFLOP/s/GPU):
2/ 220.1 (TFLOP/s/GPU):
3/ 223.2 (TFLOP/s/GPU):
4/ 224.1 (TFLOP/s/GPU):
5/ 224.3 (TFLOP/s/GPU):
6/ 224.6 (TFLOP/s/GPU):
7/ 224.1 (TFLOP/s/GPU):
8/ 224.7 (TFLOP/s/GPU):
9/ 224.9 (TFLOP/s/GPU):
10/ 224.9 (TFLOP/s/GPU):
```
Why not stopping after the 2d one to reduce walltime ? + making the sqfs from docker adds an extra ~6 min.

Reply:
Because depending on the run it takes more steps to reach the peak performance.